services/storage/src/simcore_service_storage/handlers_files.py

```diff
@@ -128,6 +201,97 @@ async def abort_upload_file(request: web.Request):
     return web.HTTPNoContent(content_type=MIMETYPE_APPLICATION_JSON)
 
 
+@routes.post(f"/{api_vtag}/locations/{{location_id}}/files/{{file_id}}:complete", name="complete_upload_file")  # type: ignore
+async def complete_upload_file(request: web.Request):
+    query_params = parse_request_query_parameters_as(StorageQueryParamsBase, request)
+    path_params = parse_request_path_parameters_as(FilePathParams, request)
+    body = await parse_request_body_as(FileUploadCompletionBody, request)
+    log.debug(
+        "received call to complete_upload_file with %s",
+        f"{path_params=}, {query_params=}",
+    )
+
+    dsm = get_dsm_provider(request.app).get(path_params.location_id)
+    # NOTE: completing a multipart upload on AWS can take up to several minutes
+    # therefore we wait a bit to see if it completes fast and return a 204
+    # if it returns slow we return a 202 - Accepted, the client will have to check later
+    # for completeness
+    task = asyncio.create_task(
+        dsm.complete_file_upload(path_params.file_id, query_params.user_id, body.parts),
+        name=create_upload_completion_task_name(
+            query_params.user_id, path_params.file_id
+        ),
+    )
+    request.app[UPLOAD_TASKS_KEY][task.get_name()] = task
+    complete_task_state_url = request.url.join(
```

Comment:
MINOR: this ``join`` is to create an absolute path, right?
**I think** you this produces already an absolute path: ``request.app.router["foo"].url_for(...).with_query(...)``, no need to add ``request.url.join(...)``

Reply:
hmm, I'm puzzled, I think I did it that way and then the final url was not the one I needed. I can double-check though

Reply:
ok I double-checked, this creates a relative url. so I need the join part.